src/dask_nested/core.py

```diff
@@ -0,0 +1,385 @@
+    def from_nested_pandas(
+        cls,
+        data,
+        npartitions=None,
+        chunksize=None,
+        sort=True,
+    ) -> NestedFrame:
+        """Returns an Nested-Dask NestedFrame constructed from a Nested-Pandas
+        NestedFrame.
+
+        Parameters
+        ----------
+        data: `NestedFrame`
+            Nested-Pandas NestedFrame containing the underlying data
+        npartitions: `int`, optional
+            The number of partitions of the index to create. Note that depending on
+            the size and index of the dataframe, the output may have fewer
+            partitions than requested.
+        chunksize: `int`, optional
+            Size of the individual chunks of data in non-parallel objects that make up Dask frames.
+        sort: `bool`, optional
+            Whether to sort the frame by a default index.
+
+        Returns
+        ----------
+        result: `NestedFrame`
+            The constructed Dask-Nested NestedFrame object.
+        """
+        result = dd.from_pandas(data, npartitions=npartitions, chunksize=chunksize, sort=sort)
+        return NestedFrame.from_dask_dataframe(result)
+
+    @classmethod
+    def from_dask_dataframe(cls, df) -> NestedFrame:
+        """Converts a Dask Dataframe to a Dask-Nested NestedFrame
+
+        Parameters
+        ----------
+        df:
+            A Dask Dataframe to convert
+
+        Returns
+        -------
+        `dask_nested.NestedFrame`
+        """
+        return df.map_partitions(npd.NestedFrame)
+
+    def compute(self, **kwargs):
+        """Compute this Dask collection, returning the underlying dataframe or series."""
+        return npd.NestedFrame(super().compute(**kwargs))
+
+    @property
+    def all_columns(self) -> dict:
+        """returns a dictionary of columns for each base/nested dataframe"""
+        all_columns = {"base": self.columns}
+        for column in self.columns:
+            if isinstance(self[column].dtype, NestedDtype):
+                nest_cols = list(self.dtypes[column].fields.keys())
+                all_columns[column] = nest_cols
+        return all_columns
+
+    @property
+    def nested_columns(self) -> list:
+        """retrieves the base column names for all nested dataframes"""
+        nest_cols = []
+        for column in self.columns:
+            if isinstance(self[column].dtype, NestedDtype):
+                nest_cols.append(column)
+        return nest_cols
+
+    def add_nested(self, nested, name) -> NestedFrame:  # type: ignore[name-defined] # noqa: F821
+        """Packs a dataframe into a nested column
+
+        Parameters
+        ----------
+        nested:
+            A flat dataframe to pack into a nested column
+        name:
+            The name given to the nested column
+
+        Returns
+        -------
+        `dask_nested.NestedFrame`
+        """
+        nested = nested.map_partitions(lambda x: pack_flat(x)).rename(name)
+        return self.join(nested, how="outer")
```

Comment:
Why is it `outer` here? Should we make it configurable?

Reply:
`outer` is here mainly to just not reject any data from either table, however I think making this configurable is a good idea. Is there a better/more intuitive default value for it?

Reply:
Made this a kwarg, but still defaulting to outer. We could default to left to follow Dask, but I'm not sure if it's the most sensible default